### PR TITLE
Merging Develop to Main

### PR DESCRIPTION
**Pull request by purplesmurf1998**

### CODE REVIEW

File: backend/controllers/commentsController.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    count: comments.length
+  });
+});
+
+// @desc        Create a new comment for a specific vehicle
+// @route       POST /api/v1/comments/vehicle/:vehicleId
+// @access      Private
+exports.createComment = asyncHandler(async (req, res, next) => {
+  
+  const vehicle = await Vehicle.findById(req.params.vehicleId);
```
[the remark this answers is not in the document]
Comment (AbdulSir) [reply backend/controllers/commentsController.js:27]: Good catch. I have removed the unused local variable in this [commit](5d7f438d1f378f1ae2992a9d6a834f70e04ed4df)

File: backend/controllers/saleController.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    // if a sale has already been created for this object return an error
+    if (!!sale_check) {
+        return next(
+            new ErrorResponse(`A sale has already been created for this vehicle`, 400)
+        );
+    }
+    // create new sale object with the data passed in the request body
+    const sale = await Sale.create(req.body);
+    //Map this created sale to the vehicle
+    const newVehicle = await Vehicle.findByIdAndUpdate(req.body.vehicle, {sale: sale._id}, {
```
[the remark this answers is not in the document]
Comment (AbdulSir) [reply backend/controllers/saleController.js:24]: Good catch. I have removed the unused local variable in this [commit](5d7f438d1f378f1ae2992a9d6a834f70e04ed4df)

File: frontend/src/modules/vehicles/VehicleProperties.vue [hunk 3]
```diff
@@ -151,28 +128,29 @@ export default {
...
+    //       // leftProperties
+    //       leftProperties.push(property);
+    //     } else {
+    //       // rightProperties
+    //       rightProperties.push(property);
+    //     }
+    //   });
+    //   this.leftProperties = leftProperties;
+    //   this.rightProperties = rightProperties;
+    // },
```
Comment (AbdulSir) [inline frontend/src/modules/vehicles/VehicleProperties.vue:153]: Remove this commented code

**Review by AbdulSir — CHANGES_REQUESTED** [review]
Remove the commented code and the 'breaks' sonarCloud is complaining about in the vehicleProperties.vue file

**Comment by TZnoj** [comment]
[SonarCloud] (https://sonarcloud.io/project/issues?id=purplesmurf1998_Autotracks&issues=AX6ngDglwvec3ceE-H9a&open=AX6ngDglwvec3ceE-H9a&pullRequest=74) is improperly flagging two functions as being duplicate implementations, despite the similarity being fetching data from the database. I'm going to label it a false positive in SonarCloud.

File: frontend/src/modules/vehicles/VehicleProperties.vue (hunk 3 above)
Comment (AnthonyChraim) [reply frontend/src/modules/vehicles/VehicleProperties.vue]: fixed in the commit https://github.com/purplesmurf1998/Autotracks/pull/79/commits/848e7d87ac066714eebcfb9b63a370fc697ff283

**Review by AbdulSir — APPROVED** [review]
Everything looks good now. We have reviewed every PR before merging it into develop. We have also resolved the bugs/security hotspots sonarCloud is complaining about. It is safe to merge this branch into main.

**Review by qandeelarshad — APPROVED** [review]
Looks good to me. The bugs sonarCloud was complaining about were resolved, so its okay to merge.

**Review by sarahGhrbl — APPROVED** [review]

**Review by TZnoj — APPROVED** [review]
As all the checks now pass and the functionality has been tested, the branch is acceptable to merge into main. The remaining codes smells are non-impactful and can be fixed in future sprints as they do not affect the security or functionality of the project.

**Pull request merged**